src/auth.py

```diff
@@ -114,16 +114,25 @@ def rq_get(request: Request, key: str) -> str:
         return j.get(key, "")
     return ""
 
-def get_facebook_public_key():
-    # Get the public key from Facebook's JWKS endpoint
-    response = requests.get(FACEBOOK_JWT_ENDPOINT)
-    jwks = response.json()
-
-    # Extract the public key in PEM format
-    key_data = jwks['keys'][0]
-    public_key_pem = jwt.algorithms.RSAAlgorithm.from_jwk(key_data)
-    
-    return public_key_pem
+@lru_cache(maxsize=1)
```

Comment:
Use `@cache` here, `@lru_cache` is not needed. But how often does the Facebook public key change? If it changes e.g. every 24 hours we need a mechanism to refetch it if the last value fetched is > 12 hours old.

Reply:
The documentation doesn’t specify a time frame for public key changes, but I believe we should refetch it every week.